### ISSUE TITLE
Compatibility with Pytorch-Elastic (and kubernetes) ?

### ISSUE THREAD

**Issue by sandys** [facebookresearch/PyTorch-BigGraph#171]
Is there a way to run Biggraph on Pytorch-Elastic ?

https://github.com/pytorch/elastic
https://aws.amazon.com/blogs/containers/fault-tolerant-distributed-machine-learning-training-with-the-torchelastic-controller-for-kubernetes/

We would like to run this on kubernetes and PyTorch-Elastic allows us to do that. https://pytorch.org/elastic/0.1.0rc2/kubernetes.html

**Comment by lw** [comment]
There's no explicit support for it, unfortunately. However, I am not familiar with TorchElastic and there may be ways to get it to work anyways. On one hand, PBG relies only on standard PyTorch collective primitives (from torch.distributed), which I guess TorchElastic supports. On the other hand, we use them in a very custom model-parallel way, which may not be the DDP-style way that TorchElastic was perhaps designed for.

In short, the only way to find out is to try. If you do so I'd be thankful if you could report your findings here!
